### PR TITLE
minimum/maximum return Option

### PR DESCRIPTION
**Pull request by yihuang**
minimum/maximum should return None when bitmap is empty, 0 and MAX_INT are valid elements.

### CODE REVIEW

File: src/bitmap/imp.rs [hunk 1]
```diff
@@ -220,7 +220,7 @@ impl Bitmap {
     ///
     /// let mut bitmap = Bitmap::create();
     /// bitmap.add(1);
-    /// bitmap.remove(1);
+    /// bitmap.remove(&1);
```
Comment (saulius) [inline src/bitmap/imp.rs]: Any reason for this change?

File: src/treemap/imp.rs [hunk 2]
```diff
@@ -161,16 +160,16 @@ impl Treemap {
...
-    /// assert_eq!(treemap.maximum(), 1000);
-    /// assert_eq!(empty_treemap.maximum(), 0);
+    /// assert_eq!(treemap.maximum(), Some(1000));
+    /// assert_eq!(empty_treemap.maximum(), None);
     /// ```
-    pub fn maximum(&self) -> u64 {
+    pub fn maximum(&self) -> Option<u64> {
         self.map
             .iter()
+            .rev()
```
Comment (saulius) [inline src/treemap/imp.rs]: Why reverse the iterator direction here?

**Comment by saulius** [comment]
I think the change makes sense. Initially I wanted to keep the `croaring` behaviour, but this is a lot more Rust-y. Just added a couple of comments before we can merge.

File: src/treemap/imp.rs (hunk 2 above)
Comment (yihuang) [reply src/treemap/imp.rs]: I think the node with big numbers are at the end of iteration?

File: src/bitmap/imp.rs (hunk 1 above)
Comment (yihuang) [reply src/bitmap/imp.rs]: Ah, it's a mistake, I was trying to change the argument of ``remove`` to reference, just like other rust collections, but then I think it's too small to consider.

Comment (saulius) [reply src/bitmap/imp.rs]: Could you please remove this change then. Thank you!

File: src/treemap/imp.rs (hunk 2 above)
Comment (saulius) [reply src/treemap/imp.rs]: That's true - makes sense!

File: src/bitmap/imp.rs (hunk 1 above)
Comment (yihuang) [reply src/bitmap/imp.rs]: Just force pushed it ;D

Comment (saulius) [reply src/bitmap/imp.rs]: It's fine, thank you!

**Pull request merged**